vars/pmebuild.groovy

```diff
@@ -74,7 +74,10 @@ def checkoutProjects(List<String> projectCollection, Map<String, Object> buildCo
                 checkoutProject(name, group, getProjectConfiguration("${group}/${name}", buildConfig), buildConfigAdditionalVariables)
             }
         } else {
-            println "[WARNING] the project won't be checked out for '${group}/${name}'"
+            println "[WARNING] '${group}/${name}' exists, cleaning Git working tree"
+            dir("${env.WORKSPACE}/${group}_${name}") {
```

Comment:
one tip: instead of repeating the same dir, you could encapsulate the whole `if` into the `dir`. wdyt ?

Reply:
Good tip, I thought about it but I think the `dir` step would create the directory if it doesn't exist and simply change the working directory if it does.

Reply:
I could probably do a `dir` to switch to the working directory and then if it's empty do the checkout and if not just clean it. Will try that.

Reply:
yeah or simply in fact put the dir name into a variable in fact and let the code like it is right now (you are using that name 3 times there)
It is just to avoid errors if the dir name changes ;)

Reply:
- Ok, I've inverted the logic to first create/switch to the Git working directory and then checkout or clean it depending on the existence of the `.git` directory inside it.
 - Extrated the checkout directory path to variable.
 - Fixed a failing test.